### DISCUSSION TITLE
How to build for visionOS?

### DISCUSSION THREAD

**Discussion by kevinmore** [curl/curl#15357] [Q&A]
Hi,

I'm recently wrapping a C libaray, which depends on CURL, into a swift package. It is then used for building a native visionOS app.

I've been following the [official document](https://curl.se/docs/install.html) for building from source. 

```
export ARCH=arm64
export SDK=xros
export DEPLOYMENT_TARGET=2.0
export CFLAGS="-arch $ARCH -isysroot $(xcrun -sdk $SDK --show-sdk-path) -m$SDK-version-min=$DEPLOYMENT_TARGET"

./configure --host=$ARCH-apple-darwin --prefix $(pwd)/artifacts --with-secure-transport
```

However, `configure` is not able to recoginize "xros". When running command:
```
xcodebuild -showsdks
```

I got the following results:
```
DriverKit SDKs:
        DriverKit 24.0                  -sdk driverkit24.0

iOS SDKs:
        iOS 18.0                        -sdk iphoneos18.0

iOS Simulator SDKs:
        Simulator - iOS 18.0            -sdk iphonesimulator18.0

macOS SDKs:
        macOS 15.0                      -sdk macosx15.0
        macOS 15.0                      -sdk macosx15.0

tvOS SDKs:
        tvOS 18.0                       -sdk appletvos18.0

tvOS Simulator SDKs:
        Simulator - tvOS 18.0           -sdk appletvsimulator18.0

visionOS SDKs:
        visionOS 2.0                    -sdk xros2.0

visionOS Simulator SDKs:
        Simulator - visionOS 2.0        -sdk xrsimulator2.0

watchOS SDKs:
        watchOS 11.0                    -sdk watchos11.0

watchOS Simulator SDKs:
        Simulator - watchOS 11.0        -sdk watchsimulator11.0
```

Attaching the config.log file below. Could someone help me with this?

Thanks!
[config.log](https://github.com/user-attachments/files/17475750/config.log)

**Comment by rinsuki** [comment]
since `-m*-version-min` is not available for visionOS, you need to replace `-m$SDK-version-min=$DEPLOYMENT_TARGET` part in CFLAGS with another things.

https://github.com/llvm/llvm-project/issues/88271

maybe something like `-mtargetos=$SDK$DEPLOYMENT_TARGET` would be work.

(actually I'm not compiling curl for visionOS but some C files for visionOS, maybe not work with curl)

**Reply by rinsuki** [reply]
or maybe you just only need specify `--target $ARCH-apple-$SDK$DEPLOYMENT_TARGET` to **`./configure`**, instead of specifying `-arch` and `-m*-version-min` option in CFLAGS (`-isysroot` part is still needed, though)

**Comment by kevinmore** [comment]
@rinsuki Thanks. I'm closing the discussion now because I just put all the CURL source files into the pacakge and managed to compile it.

**Comment by vszakats** [comment]
Here we received an Issue report with iOS build instructions with `./configure`.
It may be translatable for VisionOS:
https://github.com/curl/curl/issues/15025#issue-2544127545
